tools/SourceKit/lib/SwiftLang/SwiftASTManager.cpp

```diff
@@ -891,3 +889,36 @@ ASTUnitRef ASTProducer::createASTUnit(SwiftASTManager::Implementation &MgrImpl,
 
   return ASTRef;
 }
+
+void ASTProducer::findSnapshotAndOpenFiles(
+    SwiftASTManager::Implementation &MgrImpl,
+    ArrayRef<ImmutableTextSnapshotRef> Snapshots,
+    SmallVectorImpl<FileContent> &Contents, std::string &Error) const {
+  const InvocationOptions &Opts = InvokRef->Impl.Opts;
+  for (const auto &input :
+       Opts.Invok.getFrontendOptions().Inputs.getAllFiles()) {
+    StringRef File = input.file();
+    bool IsPrimary = input.isPrimary();
+    bool FoundSnapshot = false;
+    for (auto &Snap : Snapshots) {
+      if (Snap->getFilename() == File) {
+        FoundSnapshot = true;
+        Contents.push_back(getFileContentFromSnap(Snap, IsPrimary, File));
+        break;
+      }
+    }
+    if (FoundSnapshot)
+      break;
```

Comment:
I just noticed that this changed from `continue` to `break` in your previous change (sorry, should have noticed in my previous review), which seems wrong.  This is trying to find snapshots for *all* of the inputs, so we don't want to stop as soon as we find a single snapshot.  I'm a bit surprised this wasn't tested somewhere already.

Reply:
Looking back at 10/4/17, before I touched the code, it was a break, not a continue. The break is only breaking out of an inner loop. Which do you think is correct? I think I had previously messed up with the continue. Thanks.

Here is the original code:

bool ASTProducer::shouldRebuild(SwiftASTManager::Implementation &MgrImpl,
                                ArrayRef<ImmutableTextSnapshotRef> Snapshots) {
  const SwiftInvocation::Implementation &Invok = InvokRef->Impl;

  // Check if the inputs changed.
  SmallVector<BufferStamp, 8> InputStamps;
  InputStamps.reserve(Invok.Opts.Invok.getInputFilenames().size());
  for (auto &File : Invok.Opts.Invok.getInputFilenames()) {
    bool FoundSnapshot = false;
    for (auto &Snap : Snapshots) {
      if (Snap->getFilename() == File) {
        FoundSnapshot = true;
        InputStamps.push_back(Snap->getStamp());
        break;
      }
    }
    if (!FoundSnapshot)
      InputStamps.push_back(MgrImpl.getBufferStamp(File));
  }
  assert(InputStamps.size() == Invok.Opts.Invok.getInputFilenames().size());
  if (Stamps != InputStamps)
    return true;

  for (auto &Dependency : DependencyStamps) {
    if (Dependency.second != MgrImpl.getBufferStamp(Dependency.first))
      return true;
  }

  return false;
}

Reply:
The original code was in `createASTManager`, not `shouldRebuild` (they've very similar, but weren't identical).  The break in the above snippet is indeed in an inner loop, but the one that was changed was not and was definitely a `continue` before.

Reply:
Sorry--my mistake! Thanks for catching this. I've fixed it and will retest.